drivers/soundwire/intel.c

```diff
@@ -1081,45 +1071,81 @@ static int intel_init(struct sdw_intel *sdw)
 }
 
 /*
- * probe and init
+ * probe, startup and remove callbacks
```

Comment:
wouldn't it be possible to split this patch into 2: one switching to scan+probe+startup, and another one converting to the new driver model?